### DISCUSSION TITLE
Why is it important to prevent guessing of users?

### DISCUSSION THREAD

**Discussion by acmpo6ou** [fastapi-users/fastapi-users#828] [Q&A] [answered]
Hi!

In the [routes docs](https://fastapi-users.github.io/fastapi-users/usage/routes/), in particular the `/forgot-password` route, you have this:

> To prevent malicious users from guessing existing users in your database, the route will always return a 202 Accepted response, even if the user requested does not exist. 

Why do we need this? The docs for `/register` say that if the user doesn't exist, a `400 Bad Request` will be returned with body:
```json
{
    "detail": "REGISTER_USER_ALREADY_EXISTS"
}
```
Doesn't that allow to guess the users?

Basically, why do we prevent guessing of the users in `/forgot-password` route but allow it in `/register`?

**Comment by frankie567** [comment, marked as answer]
Hi @acmpo6ou!

Yes, I agree with you that it sounds a bit useless if the register endpoint returns this information.

TBH, I half followed the OWASP recommendations which tells us to not give any clue to a potential attacker if an account exists or not (https://cheatsheetseries.owasp.org/cheatsheets/Forgot_Password_Cheat_Sheet.html#forgot-password-request and https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html#account-creation).

I did it for forgot password but not for register. I have to think about a better way for this.

**Comment by acmpo6ou** [comment]
Ah, OK. I was just curious. Thanks!
